R/transformAbundance.R

```diff
@@ -0,0 +1,400 @@
+#' Transform Abundance
+#'
+#' This function applies transformation to abundance table, which is returned.
+#' Additionally, the information can be directly stored in the \code{assay}.
+#'
+#' @param x
+#' A \code{\link[SummarizedExperiment:SummarizedExperiment-class]{SummarizedExperiment}}
+#' object.
+#'
+#' @param abund_values
+#' A single character value for selecting the
+#'   \code{\link[SummarizedExperiment:SummarizedExperiment-class]{assay}}
+#'   to be transformed.
+#'
+#' @param transform
+#' A single character value for selecting the transformation method.
+#'
+#' @param name
+#' A single character value specifying the name of transformed abundance table.
+#'
+#' @param pseudocount
+#' Boolean or numeric value deciding whether pseudocount is added. Numerical
+#' value specifies the value of pseudocount. When selecting pseudocount==TRUE,
+#' pseudocount is 1.
+#'
+#' @param scale
+#' A numeric value for scaling. Values are multiplied with specified value.
+#'
+#' @param target
+#' A single character value specifying, if z transformation is applied
+#' for samples or features.
+#'
+#' @details
+#' \code{transformAbundance} applies transformation to abundance table.
+#' Transformation methods include
+#'
+#' \itemize{
+#'
+#'     \item{"relabundance" }{Transforms abundances to relative.
+#'     ($\frac{x}{x_{tot}}$, where $x$ is a single value and $x_{tot}$ is the sum of
+#'     all values.)}
+#'
+#'     \item{"Z" }{Z-transform or Z-standardization can be used for normalizing the data.
+#'     It can be done for samples or features by specifying 'target'. However, Z-transform
+#'     done for features can give misleading results. By default, it is done for samples.
+#'     ($\frac{x + µ}{σ}$, where $x$ is a single value, $µ$ is the mean of the sample, and
+#'     $σ$ is the standard deviation of the sample.)}
+#'
+#'     \item{"log10" }{log10 transform can be used for reducing the skewness of the data.
+#'     ($log_{10}x$, where $x$ is a single value of data.)}
+#'
+#'     \item{"log10p }{log10 transform can be used for reducing the skewness of the data.
+#'     With log10p, pseudo count is added before log10 transformation.
+#'     ($log_{10}x$, where $x$ is a single value of data.)}
+#'
+#'     \item{"hellinger" }{Hellinger transform can be used for reducing the impact of
+#'     extreme data points. ($\sqrt{\frac{x}{x_{tot}}}$, where $x$ is a single value and $x_{tot}$ is the sum of
+#'     all values.)}
+#'
+#'     \item{"identity" }{Identity is the data itself without transform.}
+#'
+#'     \item{"clr" }{Centered log ratio (clr) transform can be used for reducing the
+#'     skewness of data and for centering it.
+#'     ($log_{10}x_{r}) - log_{10}µ_{r}$, where $x_{r}$ is a single relative value, $µ_{r}$ is
+#'     the mean of relative values of whole sample.}
+#'
+#' }
+#'
+#' @references
+#' If there is references Add information here......................................
```

Comment:
Cite [Gloor et al. ](https://www.frontiersin.org/articles/10.3389/fmicb.2017.02224/full) for compositional and clr transformations. In addition, you could cite the original references to CLR (check Gloor et al for that).

Also Hellinger transformation good to cite. Perhaps vegan package has the reference if you do not find otherwise. Ask if not. 

Others are standard, no need to cite anything specifically I think.

Reply:
Added references